packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/ClassInfoModel.java

```diff
@@ -299,12 +299,20 @@ public List<MethodInfoModel> getMethods() {
         return methods;
     }
 
-    public PackageInfoModel getPackage() {
-        if (pkg == null) {
-            pkg = preparePackage();
+    /**
+     * Tries to find all packages that are ancestors of this class. From the
+     * list of all possible ancestor packages, only those whose existence is
+     * confirmed are returned. This is a "best effort" to find them as, even if
+     * they exist, there is no guarantee that they have already been loaded.
+     *
+     * @return a list of packages that are ancestors of this class
+     */
+    public List<PackageInfoModel> findAllAvailableAncestors() {
```

Comment:
Oh, almost missed. I guess, there should be not `ancestors` but `packages` because it is `ClassInfoModel`, not `PackageInfoModel`. Also, I guess, it would be better to have it as a getter. Not sure, but for me `get` is more clear than `find`.

Like:
```suggestion
    public List<PackageInfoModel> getAllAvailablePackages() {
```

Reply:
There's no obligation to use the `get` prefix to access something. This is not a property. The reason to use "find" was to make more explicit that here we're searching for something that could be even not found. I'd keep the "find" prefix here.

Concerning the "ancestors", there's no notion of packages (plural) for a class, so while `getPackage` makes sense to me, `getPackages` doesn't.

I'd rather simplify to `findAncestors` (no need to mention `packages` as it can be inferred by the returned type).

Reply:
Wouldn't users be confused if it will return a package but not the parent class (in case of a nested class)?

Reply:
I don't think so, given that the return type is `List<PackageInfoModel>`.

Reply:
Hmm, ok, sounds reasonable. I agree with having it shorter then:
```suggestion
    public List<PackageInfoModel> findAncestors() {
```